src/OrchardCore.Modules/OrchardCore.Queries/Sql/SqlQuerySource.cs

```diff
@@ -84,33 +103,31 @@ public async Task<IQueryResults> ExecuteQueryAsync(Query query, IDictionary<stri
                     else
                     {
                         column = rowDictionary.FirstOrDefault(kv => kv.Value is long).Key
-                            ?? rowDictionary.First().Key;
+                                 ?? rowDictionary.First().Key;
                     }
                 }
 
-                return rowDictionary.TryGetValue(column, out var documentIdObject) && documentIdObject is long documentId
-                    ? documentId
-                    : 0;
-            }).ToArray();
-
+                return (rowDictionary.TryGetValue(column, out var documentIdObject) ? documentIdObject : null) switch
+                {
+                    long longValue => longValue,
+                    ulong unsignedLongValue => (long)unsignedLongValue,
```

Comment:
YesSql only supports int and long. Casing ulong to long could cause an exception. the same when casting uint to int.

Reply:
> YesSql only supports int and long.

Ok, I'll remove it if it's impossible.

> Casing ulong to long could cause an exception. the same when casting uint to int.

Casting from `ulong` to `long` can only cause an exception if the number is larger than `long.Max` (9,223,372,036,854,775,807) which is quite unrealistic for an index. Also the code is casting `uint` to `long`, not `int`. That's guaranteed safe.

Reply:
Yeah I get that. But YesSql does not support the unsigned version. so supporting it here is not a good idea because the selected column could be from an unassigned value. Meaning I can do `SELECT CustomLongColumn AS DocumentId FROM CustomTable` and that would throw exception if any returned value is > long.MaxValue.

Reply:
In the very specific scenario of `SELECT CustomLongColumn AS DocumentId FROM CustomTable` the query is bad so throwing an exception is completely appropriate. Also [I've already removed](https://github.com/OrchardCMS/OrchardCore/pull/16581/commits/d6e2dd16fe3d162c0a34bdb512656ced9e9b4efb) the unsigned cases.